flags.cmake

```diff
@@ -0,0 +1,201 @@
+include(CheckCCompilerFlag)
+include(CheckCXXCompilerFlag)
+include(CMakePushCheckState)
+
+# Validate
+if(ENABLE_RCT2)
+    # We require certain settings for using the original executable
+    if(WIN32)
+        message(SEND_ERROR "ENABLE_RCT2 is not supported on Windows")
+    endif()
+    if(NOT CMAKE_SIZEOF_VOID_P EQUAL 4)
+        message(SEND_ERROR "ENABLE_RCT2 requires 32-bit compiler")
+    endif()
+endif()
+
+# Build an interface target to set all the compile options
+add_library(openrct2-flags-iface INTERFACE)
+add_library(openrct2::flags ALIAS openrct2-flags-iface)
+
+cmake_push_check_state(RESET)
+
+# Check for required compilation flags
+if(NOT MSVC)
+    set(required_flags
+        -fstrict-aliasing
+        -Werror # Warnings as errors
+        -Wundef
+        -Wmissing-declarations
+        -Winit-self
+        -Wall
+        -Wno-unknown-pragmas
+        -Wno-unused-function
+        -Wno-missing-braces
+        -Wno-comment
+        -Wshadow
+        -Wnonnull
+        -fstrict-overflow
+        )
+    set(maybe_flags
+        -Wno-error=date-time  # Launchpad turns on -Wdate-time for compilers that support it, this shouldn't break our build
+        -Wno-error=write-strings
+        -Wno-error=redundant-decls
+        -Wredundant-decls
+        -Wignored-qualifiers
+        -Wnull-dereference
+        $<$<COMPILE_LANGUAGE:CXX>:-Wsuggest-final-types>
+        $<$<COMPILE_LANGUAGE:CXX>:-Wsuggest-final-methods>
+        $<$<COMPILE_LANGUAGE:CXX>:-Wsuggest-override>
+        $<$<COMPILE_LANGUAGE:CXX>:-Wnon-virtual-dtor>
+        -Wduplicated-cond
+        -Wstrict-overflow=1
+        -Wimplicit
+        )
+    set(maybe_cxx_flags)
```

Comment:
What's this? I see C++ flags already handled in `maybe_flags` so is this still needed?

Reply:
Accident